libbeat/kibana/transform_test.go

```diff
@@ -283,3 +283,71 @@ func TestTransformFieldFormatMap(t *testing.T) {
 		assert.Equal(t, test.expected, out, fmt.Sprintf("Failed for idx %v", idx))
 	}
 }
+func TestTransformGroupAndEnabled(t *testing.T) {
+	tests := []struct {
+		commonFields common.Fields
+		expected     []string
+	}{
+		{
+			commonFields: common.Fields{common.Field{Name: "context", Path: "something"}},
+			expected:     []string{"context"},
+		},
+		{
+			commonFields: common.Fields{
+				common.Field{
+					Name: "context",
+					Type: "group",
+					Fields: common.Fields{
+						common.Field{Name: "type", Type: ""},
+						common.Field{
+							Name: "metric",
+							Type: "group",
+							Fields: common.Fields{
+								common.Field{Name: "object"},
+							},
+						},
+					},
+				},
+			},
+			expected: []string{"context.type", "context.metric.object"},
+		},
+		{
+			commonFields: common.Fields{
+				common.Field{Name: "enabledField"},
+				common.Field{Name: "disabledField", Enabled: &falsy}, //enabled is ignored for Type!=group
```

Comment:
Makes sense. I wonder if for this case we should even return an error as we already know it's false? Or is this only for Kibana?

Reply:
This is only for kibana